src/main/kotlin/no/fdk/dataset/preview/service/PreviewService.kt

```diff
@@ -39,54 +38,79 @@ class PreviewService(
         }
     }
 
-    @Throws(Exception::class)
-    private fun readCSV(resourceUrl: String, maxNumberOfRows: Int): Table {
-        if(LOGGER.isDebugEnabled) {
-            LOGGER.debug("Reading CSV for $resourceUrl")
+    private fun getMaxNumberOfRows(rows: Int?): Int {
+        if(rows == null) {
+            return DEFAULT_ROWS
         }
+        return if(rows > MAX_ROWS) MAX_ROWS else rows
+    }
+
+    fun readAndParseResource(resourceUrl: String, rows: Int?): Preview {
+        logDebug("Read and parse resource $resourceUrl")
 
-        val body = downloader.download(resourceUrl)
-        if(!isCsv(body.contentType()?.subtype)) {
-            throw Exception("Invalid content type ${body.contentType().toString()}")
+        try {
+            val body = downloader.download(resourceUrl)
+            if(isCsv(body.contentType())) {
```

Comment:
Flytt parse-koden ut til egne funksjoner og naviger her med en enkel when, feks
```suggestion
            return when {
                isCsv(body.contentType()) -> csvPreview(resourceUrl, rows, body)
                isPlain(body.contentType()) -> plainPreview(body)
                else -> throw PreviewException("Invalid content type ${body.contentType().toString()}")
            }
```